add dotpay milestone 1

AdaLam7 opened the pull request:
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/715

Noc2 commented:
Thanks for the delivery. We will look into it as soon as possible.

Noc2 reviewed (changes requested):
I finally started to look into your delivery. You can find my evaluation [here](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/DotPay_1_Noc2.md). Could you take a look at it and let me know in case you have any questions?

AdaLam7 commented:
Thanks for your feedback, we will make changes based on your suggestions.

AdaLam7 commented:
Sorry for the previous delivery issue. We have updated based on your comments.

> 0b Can’t access the documentation or see anything without login via github and sharing my data.

Now you don't need login to access the docs.

> Only description on how to run tests inside docker.

Add test from source code, and we fixed docker test failed issue.

> Docker set up doesn't seem to work for me, see below. http://bytepay.local-dev.host/ doesn't open

We update the README, and now you can use http://localhost:10086/ to access local website.

> 08.02.22: Test fail, the contract seems mostly to be a copy of [contract-transfer](https://github.com/paritytech/ink/blob/ba7e8edbae4a3dd8460b37d4ee30cf31f00a2fc3/examples/contract-transfer/lib.rs). The documentation should be updated at least

We update the contract documentation. But contract has two unit test failed caused by canvas UI and ink itself, see more detail about issue: https://github.com/paritytech/ink/issues/1117

AdaLam7 commented:
We have fixed all deliverables~ please check out 😃 ~

sulnong commented:
Notice: Please pull code before test~

Noc2 reviewed (changes requested):
Sorry for the delay here and thanks for the changes. I just updated [my evaluation](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/DotPay_1_Noc2.md). I was now able to test the docker version, but it would be nice to also have a way to set up everything locally. For example, the docker version showed me a substrate address, but I’m not sure how I can connect to the chain and fund this substrate address for testing. Additionally, I found it strange that the live version already works with DOTs. This means the smart contract is not used or can you point to the place where it’s integrated. Also where in the backend is the data stored and the address generated? Could you share links to the actual implementation with me? Thanks!

AdaLam7 commented:
@Noc2 thanks for the review.

>  it would be nice to also have a way to set up everything locally

We will provide it soon~

>  I’m not sure how I can connect to the chain and fund this substrate address for testing

Now we use Westend for testing now, the local chain only tests for the contract.

> This means the smart contract is not used 

* Yes, the live version works now but not used contract now. Because the Polkadot parallel chain does not currently support ink!. Now the contract can only run tests individually and will be integrated in the future.

![image](https://user-images.githubusercontent.com/94216827/154187345-0a07f303-c58e-4287-a8dc-eeb394750dd4.png)

* The Polkadot ink! contract test local has some issue caused by ink itself and canvas, see [know issue](https://github.com/bytepayment/bytepay#ink-transfer-contract)

> Also wherein the backend is the data stored and the address generated? 

All the data is now stored in our MongoDB, and we back up the database in real-time. The most important data is the project owner mnemonic, we store it now and will disable it when ink! works in feature. 

[code link](https://github.com/bytepayment/bytepay/blob/main/cloudfuncs/get_polkdot_keyring/index.ts)

By the way, the bytepay project is developed on faas platform named [laf](https://github.com/lafjs/laf) which is our own open-source project.

AdaLam7 commented:
![image](https://user-images.githubusercontent.com/94216827/154189599-0a2258f6-d0ab-4616-a7ef-734692d944de.png)

1. Save the owner account info in the database now, not save developer mnemonic if they bind address. [keyring code link](https://github.com/bytepayment/bytepay/blob/main/cloudfuncs/get_polkdot_keyring/index.ts)
2. [webhook source code](https://github.com/bytepayment/bytepay/blob/main/cloudfuncs/webhooks/index.ts) 
3. [Adress binding source code](https://github.com/bytepayment/bytepay/blob/main/cloudfuncs/bind_polka_address/index.ts)
4. The live version is polkadot address. Maybe caused by the docker image not upgrade, we will update it.
![image](https://user-images.githubusercontent.com/94216827/154190393-933c9db9-2647-48a5-9c5b-602885f16372.png)
Just transfer to the account to recharge like using QRcode: 
![image](https://user-images.githubusercontent.com/94216827/154190472-4f068e23-26b4-4ada-9499-19490fdfbb45.png)

Noc2 reviewed (changes requested):
Thanks for the super quick reply here and providing the links to the code. Makes sense. Let me know once you added a description on how to run things locally without docker and you updated the docker image. Ideally at least the local version would run with the smart contract, but I understand that this currently doesn’t seem to be your fault. Therefore it won’t be necessary for the approval of this milestone.

AdaLam7 commented:
> Shows a substrate account instead of a polkakdot address, not sure how to fund this account

We already fixed it and update the docker image.

Also now you can run frontend and backend using docker-compose: https://github.com/bytepayment/bytepay#run-all-in-local-mode it also contains mongo.  But the project also needs Interact with github，so the webhook will not work because github can't access your local server.

> added a description on how to run things locally without docker

We have a lot of dependencies, it might be more complicated not to use docker, is it possible to provide a Dockerfile to describe the way to run locally, since the two are almost the same.

Noc2 commented:
Thanks again for the update. A transparent Dockerfile that describes the way to run it locally is fine. But when I run “ docker-compose up”, I currently get the following error: 

<pre>sudo docker-compose up
Starting bytepay_bytepay_1 ... <font color="#26A269">done</font>
Attaching to bytepay_bytepay_1
<font color="#2AA1B3">bytepay_1  |</font> /docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
<font color="#2AA1B3">bytepay_1  |</font> /docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
<font color="#2AA1B3">bytepay_1  |</font> /docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
<font color="#2AA1B3">bytepay_1  |</font> 10-listen-on-ipv6-by-default.sh: info: IPv6 listen already enabled
<font color="#2AA1B3">bytepay_1  |</font> /docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
<font color="#2AA1B3">bytepay_1  |</font> /docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
<font color="#2AA1B3">bytepay_1  |</font> /docker-entrypoint.sh: Configuration complete; ready for start up
<font color="#2AA1B3">bytepay_1  |</font> 2022/02/17 12:55:28 [emerg] 1#1: cannot load certificate &quot;/cert/bytepay.online.fullchain.pem&quot;: BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen(&apos;/cert/bytepay.online.fullchain.pem&apos;,&apos;r&apos;) error:2006D080:BIO routines:BIO_new_file:no such file)
<font color="#2AA1B3">bytepay_1  |</font> nginx: [emerg] cannot load certificate &quot;/cert/bytepay.online.fullchain.pem&quot;: BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen(&apos;/cert/bytepay.online.fullchain.pem&apos;,&apos;r&apos;) error:2006D080:BIO routines:BIO_new_file:no such file)
<font color="#2AA1B3">bytepay_bytepay_1 exited with code 1</font>
</pre>

sulnong commented:
@Noc2 
Seems this error is caused by old version compose file.
We update compose file,  please pull the code and check again  :)

Noc2 reviewed (approved):
Indeed, it works now. I can confirm that the milestone is a pass and I will forward your invoice internally.

fededubbi commented:
Good afternoon,

I noticed that on the invoice is missing your address.
Can you please send us your invoice updated with your address?

Many thanks,
Federica Dubbini

Noc2 commented:
Thanks for the suggestions, I will bring it up internally. But we might have reservations about it because of the centralization of it right now and our security team probably would require us to host it ourselves.

AdaLam7 commented:
> Indeed, it works now. I can confirm that the milestone is a pass and I will forward your invoice internally.

@Noc2 We have submitted the invoice via the form, it has been two weeks, and have not received the M1 funding, just wondering if we are missing something？

Noc2 commented:
Thanks for chasing us. I just pinged the invoice team about it.

RouvenP commented:
hi @AdaLam7 apologies for the delay. The transfer will be executed tomorrow. We will notify you as soon as done.

RouvenP commented:
hi @AdaLam7, we transferred the payment today. Thanks!
